nifi-nar-bundles/nifi-compress-bundle/nifi-compress-processors/src/main/java/org/apache/nifi/processors/compress/ModifyCompression.java

```diff
@@ -0,0 +1,420 @@
+            .defaultValue("1")
+            .required(true)
+            .allowableValues("0", "1", "2", "3", "4", "5", "6", "7", "8", "9")
+            .dependsOn(OUTPUT_COMPRESSION,
+                    CompressionInfo.COMPRESSION_FORMAT_ATTRIBUTE,
+                    CompressionInfo.COMPRESSION_FORMAT_GZIP,
+                    CompressionInfo.COMPRESSION_FORMAT_DEFLATE,
+                    CompressionInfo.COMPRESSION_FORMAT_XZ_LZMA2,
+                    CompressionInfo.COMPRESSION_FORMAT_ZSTD,
+                    CompressionInfo.COMPRESSION_FORMAT_BROTLI)
+            .build();
+
+    public static final PropertyDescriptor UPDATE_FILENAME = new PropertyDescriptor.Builder()
+            .name("Update Filename")
+            .description("If true, will remove the filename extension when decompressing data (only if the extension indicates the appropriate "
+                    + "compression format) and add the appropriate extension when compressing data")
+            .required(true)
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("FlowFiles will be transferred to the success relationship after successfully being compressed or decompressed")
+            .build();
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("FlowFiles will be transferred to the failure relationship if they fail to compress/decompress")
+            .build();
+
+    private List<PropertyDescriptor> properties;
+    private Set<Relationship> relationships;
+    private Map<String, String> compressionFormatMimeTypeMap;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(INPUT_COMPRESSION);
+        properties.add(OUTPUT_COMPRESSION);
+        properties.add(COMPRESSION_LEVEL);
+        properties.add(UPDATE_FILENAME);
+        this.properties = Collections.unmodifiableList(properties);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        this.relationships = Collections.unmodifiableSet(relationships);
+
+        final Map<String, String> mimeTypeMap = new HashMap<>();
+        for(CompressionInfo compressionInfo : CompressionInfo.values()) {
+            String[] mimeTypes = compressionInfo.getMimeTypes();
+            if (mimeTypes == null) {
+                continue;
+            }
+            for(String mimeType : mimeTypes) {
+                mimeTypeMap.put(mimeType, compressionInfo.getValue());
+            }
+        }
+
+        this.compressionFormatMimeTypeMap = Collections.unmodifiableMap(mimeTypeMap);
```

Comment:
We still need to have entries in a map or otherwise for the multiple possible MIME types for a single CompressionInfo, I moved this to a static initializer so it's only done once.